src/source/PeerConnection/SessionDescription.c

```diff
@@ -155,12 +155,14 @@ STATUS setPayloadTypesFromOffer(PHashTable codecTable, PHashTable rtxTable, PSes
     PSdpMediaDescription pMediaDescription = NULL;
     UINT8 currentMedia, currentAttribute;
     PCHAR attributeValue, end;
-    UINT64 parsedPayloadType, rtxPayloadType, hashmapPayloadType;
+    UINT64 parsedPayloadType, rtxPayloadType, hashmapPayloadType, fmtpVal, aptVal, aptFmtVal;
+    PSingleList pListAptFmtpVals = NULL;
     BOOL supportCodec;
-    UINT32 tokenLen;
+    UINT32 tokenLen, nodeCount, i;
 
     for (currentMedia = 0; currentMedia < pSessionDescription->mediaCount; currentMedia++) {
         pMediaDescription = &(pSessionDescription->mediaDescriptions[currentMedia]);
+        singleListCreate(&pListAptFmtpVals);
```

Comment:
There is a handful of these values. You could define a fixed array of MAX_FMTP_VALUES and a count and set the values in the array. You could have two arrays to store the the fmtp value and the type in each for example. It will be way faster than the list and will be almost the same size if there are only a few of these values expected.

Reply:
I really wanted to do this but it was taking too long to find what MAX_FMTP_VALUES should be without using too much memory.  Now that you mention it as as well I'm going to use this, currently the max SDP attributes is 128 sdp attributes and each rtx type requires 2 attributes so 64 is a safe value for MAX_FMTP_VALUES, I'm going to go with that.